src/lambdaisland/embedkit.clj

```diff
@@ -86,7 +88,22 @@
          :content-type :json
          :as :json}
         (cond-> middleware (assoc :middleware middleware))
-        (into opts))))
+        (into opts)))
+  (conn-cache [this]
+    (:cache this))
+  (secret-key [this]
+    (:secret-key this)))
+
+(extend-protocol IConnection
+  clojure.lang.IAtom
```

Comment:
If we get an Atom where we expect a connection, simply delegate all methods to the connection that is assumed to be inside the atom.